syntax = "proto3";

package Auth;

service Authenticate {
    rpc authenticate(AuthRequest) returns (AuthResponse) {}
    rpc validate(JWTValidateRequest) returns (JWTValidateResponse) {}
    rpc invalidate(JWTValidateRequest) returns (JWTInvalidateResponse) {}
}

message AuthRequest {
    string username = 1;
    string password = 2;
}

message AuthResponse {
    string status = 1;
    string message = 2;
}

message JWTValidateRequest {
    string jwt = 1;
}

message JWTValidateResponse {
    bool valid = 1; 
}

message JWTInvalidateResponse {
    bool deleted = 1;
}